libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt

```diff
@@ -244,7 +307,32 @@ fun List<NavigationRoute>.toDirectionsRoutes() = map { it.directionsRoute }
  *
  * **Avoid using this mapper and instead try using APIs that accept [NavigationRoute] type where possible.**
  */
-fun List<DirectionsRoute>.toNavigationRoutes() = map { it.toNavigationRoute() }
+@Deprecated(
+    "NavigationRoute requires routerOrigin. If RouterOrigin is not set uses RouterOrigin.Custom()",
+    ReplaceWith("toNavigationRoutes(routerOrigin)")
+)
+fun List<DirectionsRoute>.toNavigationRoutes() =
+    map { it.toNavigationRoute(RouterOrigin.Custom()) }
```

Comment:
Wondering if we can be smarter about it. This is a compatibility function so I think we should make the best effort to figure out the source ourselves and only fallback to `Custom` when needed.

For example, if `route.uuid` is available, it's an `offboard` route.

Reply:
I've already sync with NN, and there's no good way to check what's route origin, it also might be Custom and we make change it with a false prediction

Reply:
That's where we have the other method that does accept origin as parameter.

This parameter-less function is here to provide our best compatibility with the old router interfaces (and we can document it this way), so I still think we should at least check the UUID.

Reply:
that's actually a good point of investigation: what if a user stored the route locally (as we do in mobile-metrics actually) and set it to the SDK with `uuid` and `RouterOrigin.OffBoard` NN will assume one as `Offboard` router with annotations and incidents and trying to update it but cannot.

Reply:
> This parameter-less function is here to provide our best compatibility with the old router interfaces (and we can document it this way), so I still think we should at least check the UUID.

We might assume the following cases:
- a route has UUID and it contains `@local` string -> route is onBoard
- a route has UUID -> offBoard
- else -> Custom

cc @LukasPaczos @mskurydin to check if it makes sense

Reply:
The UUID is not well documented and its contents is subject to change. Not sure we can rely on it for this.

Reply:
As discussed internally, let's merge as is. The only thing we'll need to remember is not make the origin a strong condition for other features :+1: